File: app/src/main/java/io/github/prabhuomkar/flare/ui/activities/MainActivity.java
package io.github.prabhuomkar.flare.ui.activities;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import io.github.prabhuomkar.flare.Constants;
import io.github.prabhuomkar.flare.R;
import io.github.prabhuomkar.flare.helpers.FragmentHelper;
import io.github.prabhuomkar.flare.helpers.PermissionsHelper;
import io.github.prabhuomkar.flare.ui.fragments.AppInfoFragment;
import io.github.prabhuomkar.flare.ui.fragments.HomeFragment;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        FragmentHelper.switchFragment(new HomeFragment(), getSupportFragmentManager(), false);
        PermissionsHelper.getPermissions(MainActivity.this);
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.menu_main, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        int id = item.getItemId();
        if (id == R.id.action_contact) {
            // Send an email to Authors/Maintainers
            Intent emailIntent = new Intent(Intent.ACTION_SENDTO, Uri.fromParts(
                    "mailto", Constants.CONTACT_EMAIL, null));
            try {
                startActivity(Intent.createChooser(emailIntent, "Contact TorchExpo"));
            } catch (android.content.ActivityNotFoundException e) {
                Toast.makeText(MainActivity.this, Constants.ERR_NO_EMAIL_CLIENT, Toast.LENGTH_SHORT).show();
            }
        } else if (id == R.id.action_help) {
            // Open GitHub Repository
            startActivity(new Intent(Intent.ACTION_VIEW, Uri.parse(Constants.HELP_URL)));
        } else if (id == R.id.action_app_info) {
            // Open App Info
            FragmentHelper.switchFragment(new AppInfoFragment(), getSupportFragmentManager(), true);
        } else if (id == android.R.id.home) {
            onBackPressed();
            return true;
        }
        return super.onOptionsItemSelected(item);
    }

    @Override
    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions,
                                           @NonNull int[] grantResults) {
        PermissionsHelper.checkPermissions(MainActivity.this,
                requestCode, permissions, grantResults);
    }

}

File: app/src/main/java/io/github/prabhuomkar/torchexpo/MainActivity.kt
package io.github.prabhuomkar.torchexpo

import android.content.ActivityNotFoundException
import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.view.Menu
import android.view.MenuInflater
import android.view.MenuItem
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.navigation.findNavController
import androidx.navigation.ui.AppBarConfiguration
import androidx.navigation.ui.navigateUp
import androidx.navigation.ui.onNavDestinationSelected
import androidx.navigation.ui.setupActionBarWithNavController

class MainActivity : AppCompatActivity() {

    private lateinit var appBarConfiguration: AppBarConfiguration

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.main_activity)
        setupNavigation()
    }

    private fun setupNavigation() {
        val navController = findNavController(R.id.nav_host_fragment)
        appBarConfiguration = AppBarConfiguration(navController.graph)
        setupActionBarWithNavController(navController, appBarConfiguration)
    }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        MenuInflater(applicationContext).inflate(R.menu.options_menu, menu)
        return super.onCreateOptionsMenu(menu)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean = when (item.itemId) {
        R.id.aboutFragment -> {
            item.onNavDestinationSelected(findNavController(R.id.nav_host_fragment))
            true
        }
        R.id.action_help -> {
            startActivity(
                Intent(
                    Intent.ACTION_VIEW,
                    Uri.parse(HELP_URL)
                )
            )
            true
        }
        R.id.action_contact -> {
            val emailIntent = Intent(
                Intent.ACTION_SENDTO, Uri.fromParts(
                    "mailto", CONTACT_EMAIL, null
                )
            )
            try {
                startActivity(Intent.createChooser(emailIntent, "Contact TorchExpo"))
            } catch (e: ActivityNotFoundException) {
                Toast.makeText(
                    this@MainActivity,
                    this.getString(R.string.err_no_email_client),
                    Toast.LENGTH_SHORT
                )
                    .show()
            }
            true
        }
        else -> super.onOptionsItemSelected(item)
    }

    override fun onSupportNavigateUp(): Boolean {
        val navController = findNavController(R.id.nav_host_fragment)
        return navController.navigateUp(appBarConfiguration) || super.onSupportNavigateUp()
    }
}
